buffer for chunked streaming breaks queries thru chproxy

0.8.0 introduced a buffer for HTTP streaming/chunked queries. This feature breaks integration with some types of ClickHouse proxies. For example a popular [chproxy](https://github.com/ContentSquare/chproxy) is affected.

The reason is simple. chproxy returns non-chunked response with a valid `content-length` and no `transfer-encoding` header.

```
urllib3.exceptions.ResponseNotChunked: Response is not chunked. Header 'transfer-encoding: chunked' is missing.

```

There's no way to walk around this buffer.

I understand that this lib is for ClickHouse and chproxy might be out of scope, but there's [some hope](https://github.com/ClickHouse/clickhouse-connect/blob/d113091d4c14cfd90ce39ee33e6fe7d44dfbe400/clickhouse_connect/common.py#L80) :)


### Steps to reproduce
1. use clickhouse-connect >= 0.8.0
2. use chproxy
3. get exception

### Expected behaviour

### Code example

```python
from clickhouse_connect.driver import create_client

def test_chproxy_connection():
    test_client = create_client(host="127.0.0.1", port=9090, username='default', password="")
    query = f"SELECT count() from system.one"
    res = test_client.query(query)
    assert res.result_rows[0][0] == 1

```

### clickhouse-connect logs

```
../../clickhouse_connect/driver/client.py:218: in query
    return self._query_with_context(query_context)
../../clickhouse_connect/driver/httpclient.py:237: in _query_with_context
    query_result = self._transform.parse_response(byte_source, context)
../../clickhouse_connect/driver/transform.py:68: in parse_response
    first_block = get_block()
../../clickhouse_connect/driver/transform.py:33: in get_block
    num_cols = source.read_leb128()
../../clickhouse_connect/driver/buffer.py:65: in read_leb128
    b = self.read_byte()
../../clickhouse_connect/driver/buffer.py:51: in read_byte
    chunk = next(self.gen, None)
../../clickhouse_connect/driver/httputil.py:232: in buffered
    chunk = next(read_gen, None) # Always try to read at least one chunk if there are any left
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <urllib3.response.HTTPResponse object at 0x12840c970>, amt = 1048576
decode_content = False

    def read_chunked(
        self, amt: int | None = None, decode_content: bool | None = None
    ) -> typing.Generator[bytes, None, None]:
        self._init_decoder()
        # FIXME: Rewrite this method and make it a class with a better structured logic.
        if not self.chunked:
>           raise ResponseNotChunked(
                "Response is not chunked. "
                "Header 'transfer-encoding: chunked' is missing."
            )
E           urllib3.exceptions.ResponseNotChunked: Response is not chunked. Header 'transfer-encoding: chunked' is missing.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,7 +9,12 @@ When creating a DBAPI Connection method using the Connection constructor or a SQ
 converts any unrecognized keyword argument/query parameter to a ClickHouse server setting. Starting in the next minor
 release (0.9.0), unrecognized arguments/keywords for these methods of creating a DBAPI connection will raise an exception
 instead of being passed as ClickHouse server settings. This is in conjunction with some refactoring in Client construction.
-The supported method of passing ClickHouse server settings is to prefix such arguments/query parameters with`ch_`.  
+The supported method of passing ClickHouse server settings is to prefix such arguments/query parameters with`ch_`.
+
+## 0.8.6, 2024-11-01
+### Bug Fixes
+- Correctly stream unchunked HTTP responses.  Fixes https://github.com/ClickHouse/clickhouse-connect/issues/417.
+- Don't use `wait_end_of_query` for any streaming requests.  Fixes https://github.com/ClickHouse/clickhouse-connect/issues/416
 
 ## 0.8.5, 2024-10-24
 ### Bug fix

--- a/clickhouse_connect/__version__.py
+++ b/clickhouse_connect/__version__.py
@@ -1,1 +1,1 @@
-version = '0.8.5'
+version = '0.8.6'

--- a/clickhouse_connect/driver/httpclient.py
+++ b/clickhouse_connect/driver/httpclient.py
@@ -482,7 +482,7 @@ class HttpClient(Client):
         See BaseClient doc_string for this method
         """
         body, params, fields = self._prep_raw_query(query, parameters, settings, fmt, use_database, external_data)
-        return self._raw_request(body, params, fields=fields, stream=True)
+        return self._raw_request(body, params, fields=fields, stream=True, server_wait=False)
 
     def _prep_raw_query(self, query: str,
                         parameters: Optional[Union[Sequence, Dict[str, Any]]],

--- a/clickhouse_connect/driver/httputil.py
+++ b/clickhouse_connect/driver/httputil.py
@@ -225,7 +225,7 @@ class ResponseSource:
             chunks = deque()
             done = False
             current_size = 0
-            read_gen = response.read_chunked(chunk_size, decompress is None)
+            read_gen = response.stream(chunk_size, decompress is None)
             while True:
                 while not done:
                     chunk = next(read_gen, None) # Always try to read at least one chunk if there are any left

--- a/clickhouse_connect/driver/transform.py
+++ b/clickhouse_connect/driver/transform.py
@@ -54,13 +54,14 @@ class NativeTransform:
                 if isinstance(ex, StreamCompleteException):
                     # We ran out of data before it was expected, this could be ClickHouse reporting an error
                     # in the response
-                    message = source.last_message
-                    if len(message) > 1024:
-                        message = message[-1024:]
-                    error_start = message.find('Code: ')
-                    if error_start != -1:
-                        message = message[error_start:]
-                    raise StreamFailureError(message) from None
+                    if source.last_message:
+                        message = source.last_message
+                        if len(message) > 1024:
+                            message = message[-1024:]
+                        error_start = message.find('Code: ')
+                        if error_start != -1:
+                            message = message[error_start:]
+                        raise StreamFailureError(message) from None
                 raise
             block_num += 1
             return result_block

